### PR TITLE
feat: Ingest Points continuously with Akka Stream

### PR DESCRIPTION
**Pull request by nMoncho**
## Proposed Changes

_Briefly describe your proposed changes:_

The current Akka Streams API requires to materialize a different stream whenever we want to write a point. Sources like Kafka never end, and thus the current API requires to materialize a different stream for each point, where a single stream would do:

```scala
val source: Source[Point, NotUsed] = fromKafka() // we get a stream that never ends
source.mapAsync(parallelism) { point =>
  Source.single(point).runWith(writeApi.writePoint)
}.runWith(Sink.foreach(_ => println("recoded point")
```

With these changes we can do:

```scala
val source: Source[Point, NotUsed] = fromKafka() // we get a stream that never ends
source.runWith(writeApi.writePoint)
```

My apologies if this PR is not following the desired approach. As an avid Akka Streams users, I think this is the expected behavior out of a Sink.

## Checklist

- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `mvn test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: client-scala/src/main/scala/com/influxdb/client/scala/internal/WriteScalaApiImpl.scala [hunk 1]
```diff
@@ -101,13 +99,12 @@ class WriteScalaApiImpl(@Nonnull service: WriteService, @Nonnull options: Influx
...
    *               if the `org` is not specified.
-   * @return the sink that accept the Data points. The `point` is considered as one batch unit.
+   * @return the sink that accept the Data points.
    */
   override def writePoint(bucket: Option[String], org: Option[String]): Sink[Point, Future[Done]] = {
     Flow[Point]
       .map(point => (point.getPrecision, Seq(new AbstractWriteClient.BatchWriteDataPoint(point, options))))
-      .map(batch => writeHttp(Some(batch._1), bucket, org, batch._2))
-      .toMat(Sink.head)(Keep.right)
+      .toMat(Sink.foreach(batch =>writeHttp(Some(batch._1), bucket, org, batch._2)))(Keep.right)
```
Comment (bednar) [inline client-scala/src/main/scala/com/influxdb/client/scala/internal/WriteScalaApiImpl.scala:107]: Just typo:

```suggestion
      .toMat(Sink.foreach(batch => writeHttp(Some(batch._1), bucket, org, batch._2)))(Keep.right)
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 

There is a minor requirement that must be be satisfy before we accept the PR:

File: client-scala/src/main/scala/com/influxdb/client/scala/internal/WriteScalaApiImpl.scala (hunk 1 above)
Comment (nMoncho) [reply client-scala/src/main/scala/com/influxdb/client/scala/internal/WriteScalaApiImpl.scala]: I completely missed that, good catch. Fixed

**Review by bednar — APPROVED** [review]
LGTM 🚀

**Pull request merged**